### PR TITLE
Ziafazal/yonk-298: organization user removal support

### PR DESCRIPTION
**Pull request by ziafazal**
This PR supports removal of user and organization link. It also has some code refactoring in tests to remove duplicate code.
@mattdrayer would you review this one?

### CODE REVIEW

**Comment by ziafazal** [comment]
@mattdrayer do you have time to review this?

**Comment by ziafazal** [comment]
@afzaledx would you please review this?

File: lms/djangoapps/organizations/views.py [hunk 1]
```diff
@@ -100,7 +101,7 @@ def users(self, request, pk):
             * view parameter can be used to get a particular data .i.e. view=ids to
             * get list of user ids
         - POST: Adds a User to an Organization
-
+        - DELETE: Removes user link to an Organization for given users in `users` param
```
Comment (afzaledx) [inline lms/djangoapps/organizations/views.py]: This migt be more consistent with the descriptions of other methods:
DELETE: Removes the user(s) given in the `users` param from an Organization.

File: lms/djangoapps/organizations/views.py [hunk 2]
```diff
@@ -152,6 +153,30 @@ def users(self, request, pk):
 
                     response_data.append(user_data)
             return Response(response_data, status=status.HTTP_200_OK)
+        elif request.method == 'DELETE':
+            user_ids = request.DATA.get('users')
+            if not user_ids:
+                return Response({"detail": _('users parameter is missing.')}, status.HTTP_400_BAD_REQUEST)
+            try:
+                user_ids = map(int, user_ids.split(','))
```
Comment (afzaledx) [inline lms/djangoapps/organizations/views.py]: `map(int, user_ids.split(','))
`
would throw an exception if a user supplies the list with a trailing comma - leaving the user wondering what was wrong. The exception would be thrown because the `split()` method would create a None item in the end of the list - and int(None) would raise an exception.

Using the `filter()` method would remove empty elements.
map(int, filter(None, user_ids.split(',')))

nit: user_ids = [ int(id) for id in filter(None, user_ids.split(',')) ] might be more pythonic.

File: lms/djangoapps/organizations/views.py [hunk 3]
```diff
@@ -152,6 +153,30 @@ def users(self, request, pk):
...
+                    "detail": _('users parameter must be comma separated list of integers.')
+                }, status.HTTP_400_BAD_REQUEST)
+
+            users_removed = 0
+            organization = self.get_object()
+            for user_id in user_ids:
+                try:
+                    user = User.objects.get(id=user_id)
+                except ObjectDoesNotExist:
+                    continue
```
Comment (afzaledx) [inline lms/djangoapps/organizations/views.py]: I am not sure about the requirements, but the count of users not removed might also make sense.

File: lms/djangoapps/organizations/tests.py [hunk 4]
```diff
@@ -103,14 +104,36 @@ def do_get(self, uri):
...
         return response
 
-    def do_delete(self, uri):
+    def do_delete(self, uri, data):
         """Submit an HTTP DELETE request"""
         headers = {
             'Content-Type': 'application/json',
             'X-Edx-Api-Key': str(TEST_API_KEY),
         }
-        response = self.client.delete(uri, headers=headers)
```
Comment (afzaledx) [inline lms/djangoapps/organizations/tests.py]: Any particular reason why client.delete() was changed to client.put(REQUEST_METHOD='DELETE') to test the DELETE method of the API call?

**Comment by afzaledx** [comment]
👍 other than the split() issue. Other things are minor.
